### ISSUE TITLE
style check failures after fork merge

### ISSUE THREAD

**Issue by AlexanderRichert-NOAA** [LLNL/spack#39176]
### Steps to reproduce

We ran into the CI style check errors below after merging Spack develop into our own fork (jcsda/spack). It looks like there were changes in the style checks (either the configuration or in flake8 itself) that a number of Spack core files were not first updated to. I get the same result for a test file (lib/spack/llnl/util/tty/log.py) using a fresh copy of develop (https://github.com/spack/spack/pull/39175), so it's not a fork-specific issue.

### Error message
```
 lib/spack/llnl/util/tty/log.py:783: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/ci.py:538: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/ci.py:888: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/config.py:433: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/database.py:155: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/multimethod.py:239: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/cmd/build_env.py:57: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/llnl/util/file_list.py:137: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/llnl/util/file_list.py:155: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/llnl/util/file_list.py:156: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/llnl/util/file_list.py:222: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/llnl/util/file_list.py:240: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/llnl/util/file_list.py:241: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/sbang.py:61: [F811] redefinition of unused 'sbang_line' from line 56
lib/spack/spack/test/spec_semantics.py:518: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/spec_semantics.py:519: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/spec_semantics.py:520: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/spec_semantics.py:521: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/variant.py:528: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/variant.py:531: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/test/variant.py:534: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/util/spack_yaml.py:71: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/version.py:761: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lib/spack/spack/version.py:795: [E721] do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
```

### Information on your system

Spack GitHub Actions

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
I think this is due to additional checks being added to `flake8`. Not sure I would classify this as an issue, since we check style on changesets in PRs. @tgamblin @adamjstewart Thoughts?

In any case, feel free to submit fixes for Spack, so it won't cause troubles in your own fork :wink:

**Comment by alalazo** [comment]
Which version of Spack are we talking about? This file for instance:
```
lib/spack/spack/version.py
```
does not exist in current `develop`

**Comment by alalazo** [comment]
Marking as not-a-but for the time being, but waiting for other people to comment to check whether we should close the issue or change the triage

**Comment by adamjstewart** [comment]
Yes, our style deps get updated and change requirements but PRs only test diffs and so we don't catch this. When black 23 was released, a lot of files needed to be changed. I did this in #35351 so that it didn't fall on new users to change in their PRs. I think someone should submit a PR to update packages for flake8 compliance.

@alalazo is it possible to add a `spack style --all` test to our release tests? It doesn't need to happen frequently, but we should avoid this in general.
